reasoner/resolution/answer/AnswerStateImpl.java

```diff
@@ -1003,6 +1003,11 @@ public P aggregateToUpstream(ConceptMap concepts) {
                 return parent().with(concepts);
             }
 
+            @Override
+            public Retrievable<P> with(ConceptMap extension) {
+                return new RetrievableImpl<>(filter, parent(), extendAnswer(extension), root());
+            }
+
```

Comment:
Needed so that we can pass from a subsumer to the resolver it subsumes without the answerstate following the normal `upstream` pattern.